include/deal.II/lac/arpack_solver.h

```diff
@@ -431,6 +431,16 @@ void ArpackSolver::solve (const MatrixType1                  &/*system_matrix*/,
       break;
     }
 
+  //Check for possible options for symmetric problems
+  Assert(!additional_data.symmetric || strcmp(which, "LR"),
```

Comment:
i think this fits better to AdditionalData class as it's essentially a constraint between  `additional_data.symmetric` and `additional_data.eigenvalue_of_interest`. 

Could you do the same with pArpack?